frame/state-trie-migration/src/lib.rs

```diff
@@ -637,8 +641,8 @@ pub mod pallet {
 			match migration {
 				Ok(_) => Ok(post_info),
 				Err(error) => {
-					Self::halt(&error);
-					Err(DispatchErrorWithPostInfo { post_info, error })
```

Comment:
similar here, we could emit an event rather than return this error

Reply:
The `Halted` event stores the error.